.github/workflows/code.yml

```diff
@@ -50,3 +57,58 @@ jobs:
         run:  |
           make generate
           if [ -n "$(git status --porcelain=v1)" ]; then git add -N .; git diff; exit 1; fi
+  unit:
+    runs-on: ubuntu-latest
+    if: success() || failure()
+    strategy:
+      matrix:
+        go:
+        - version: "1.19"
+          name: target
+        - version: "1.20"
+          name: latest
+    name: "Unit tests with ${{ matrix.go.name }} Go"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go.version }}
+      - name: run unit tests
+        run: make test
+      - name: Upload coverage to Codecov
```

Comment:
could we add the integration test run here, after unit tests but before uploading coverage? Could make some coverage problems go away if we can upload the reports for "target go" and "latest go" in one go each